From fde426dbba94d841532bd42d491a4665503bba50 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 9 Mar 2023 20:11:09 +0100
Subject: [PATCH 091/111] arm64: dts: imx8mp: Add DeWarp Engine DT node

Add DT node for the DeWarp Engine of the i.MX8MP.

Upstream-Status: Submitted [https://lore.kernel.org/linux-arm-kernel/20230515163224.70300-1-marex@denx.de/]
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm64/boot/dts/freescale/imx8mp.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp.dtsi b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
index 89853f66cf61c..3d73e883c7968 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
@@ -1458,6 +1458,16 @@ isi_in_1: endpoint {
 				};
 			};
 
+			dewarp: dwe@32e30000 {
+				compatible = "nxp,imx8mp-dw100";
+				reg = <0x32e30000 0x10000>;
+				interrupts = <GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clk IMX8MP_CLK_MEDIA_AXI_ROOT>,
+					 <&clk IMX8MP_CLK_MEDIA_APB_ROOT>;
+				clock-names = "axi", "ahb";
+				power-domains = <&media_blk_ctrl IMX8MP_MEDIABLK_PD_DWE>;
+			};
+
 			mipi_csi_0: csi@32e40000 {
 				compatible = "fsl,imx8mp-mipi-csi2", "fsl,imx8mm-mipi-csi2";
 				reg = <0x32e40000 0x10000>;
-- 
2.42.0

